lib/file.c

```diff
@@ -419,10 +421,12 @@ static CURLcode file_do(struct Curl_easy *data, bool *done)
         return result;
     }
 
-    result = Curl_client_write(data, CLIENTWRITE_HEADER,
-                               (char *)"Accept-ranges: bytes\r\n", 0);
-    if(result)
-      return result;
+    if(!S_ISDIR(statbuf.st_mode)) {
+      result = Curl_client_write(data, CLIENTWRITE_HEADER,
+                                 (char *)"Accept-ranges: bytes\r\n", 0);
+      if(result)
```

Comment:
nit: `result != CURLE_OK)`

Reply:
Hmm, this won't line up with any of the other result checks in this file anymore :(

(Done)